File: src/ffi/alerts/torrent_error.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type torrent_error_alert = crate::ffi::ffi::torrent_error_alert;
    }
    unsafe extern "C++" {
        include!("cpp/alerts/torrent_error.h");
        type Error = crate::ffi::error::ffi::Error;

        unsafe fn torrent_error_alert_get_filename<'a>(alert: *mut torrent_error_alert) -> &'a str;
        unsafe fn torrent_error_alert_get_error(alert: *mut torrent_error_alert) -> Error;
    }
}

File: cpp/alerts/torrent_error.cpp
#include "./torrent_error.h"
#include "cpp/error.h"

namespace ltrs {
    rust::Str torrent_error_alert_get_filename (lt::torrent_error_alert* alert) {
        return alert->filename();
    }

    Error torrent_error_alert_get_error (lt::torrent_error_alert* alert) {
        return error_code_to_error(alert->error);
    }
}
